src/custom/hooks/useTransactionConfirmationModal.tsx

```diff
@@ -0,0 +1,40 @@
+import { useState, useCallback } from 'react'
+import TransactionConfirmationModal from 'components/TransactionConfirmationModal'
+import { OperationType } from 'components/TransactionConfirmationModal'
+import { useOpenModal, useCloseModals, useModalOpen } from 'state/application/hooks'
+import { ApplicationModal } from 'state/application/reducer'
+
+export default function useTransactionConfirmationModal(
```

Comment:
So this is just to use this hook, and allow to open the confirmation with different operation types and messages for the same component? 

im kind of ok, although i don't see completly the point since you only used it with one type, so probably using `TransactionConfirmationModal` would be not that bad

Reply:
yeah the code inside SwapMod, [these lines here + the component](https://github.com/gnosis/cowswap/blob/ddf8ca6074311d3c334418cc3dacc809a9dcfd7f/src/custom/pages/Swap/SwapMod.tsx#L152) are identical in claim. and likely anywhere else with approve/transaction modal stuff